### PR TITLE
Acceleration related changes and minor fixes

### LINKED ISSUE #127: [FEATURE] Table Acceleration Support
Flint Spark extension to create skipping index with OpenSearch which can accelerate query with applicable filtering condition. 

### Skipping Index 
* User can only create, view or delete  a skipping index 
* One table can only have 1 skipping index for a Flint table
* Name is auto-generated for a given table and follows a pre-defined by a naming convention
* Columns can be accelerated individually using partition, value_set and min_max skipping type
* This is always auto-refreshed or manually refreshed. 
### Covering Index
* User can only create, view or delete a covering index 
* User can have multiple covering index for a Flint table 
* User can define their own name for the covering Index
* User can select columns required for creating the covering index
* This is always auto-refreshed or manually refreshed. 
### Materialized View
* User can only create, view or delete a materialised view 
* User can have multiple materialized views for a Flint table 
* User can define their own name for the materialized view
* Columns can be aggregated into metrics with different functions like count, min, max, sum and avg.  
* The above metrics can be further grouped into different columns [Ex. TUMBLE() or SLIDE()]
* Final check for MVs are where clauses to restrict metrics on a given subset of data
* This is always auto-refreshed or manually refreshed.

### PR DESCRIPTION
**Pull request by ps48**
### Description
1. Add acceleration index flyout when user's click on sidebar
2. Remove [if not exists] from the query builders
3. Update router to hash router to support redirection with datasource params
4. Hide Materialized View until Opensearch-spark has a support for it
5. Backend integ to load data in acceleration flyout
6. Update default primary count to be 1 and replica count validation to allow 0
7. Remove caution callout from acceleration flyout
8. Support redirection for acceleration flyout

Pending: Load table fields from spark sql `describe table`

### Issues Resolved
#127 
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: public/components/acceleration/selectors/source_selector.tsx [hunk 1]
```diff
@@ -3,101 +3,110 @@
...
+        setLoadingComboBoxes({ ...loadingComboBoxes, database: false });
+      });
+    });
+  };
+
+  const loadTables = () => {
+    setLoadingComboBoxes({ ...loadingComboBoxes, dataTable: true });
+    const query = {
+      lang: 'sql',
+      query: `SHOW TABLES IN ${accelerationFormData.dataSource}.${accelerationFormData.database}`,
```
Comment (joshuali925) [inline public/components/acceleration/selectors/source_selector.tsx:81]: should table name be wrapped in backticks?

File: public/components/acceleration/visual_editors/query_builder.tsx [hunk 2]
```diff
@@ -74,8 +73,7 @@ const buildSkippingIndexColumns = (skippingIndexQueryData: SkippingIndexRowType[
 export const skippingIndexQueryBuilder = (accelerationformData: CreateAccelerationForm) => {
   const { dataSource, database, dataTable, skippingIndexQueryData } = accelerationformData;
 
-  const codeQuery = `CREATE SKIPPING INDEX 
-[IF NOT EXISTS]
```
Comment (joshuali925) [inline public/components/acceleration/visual_editors/query_builder.tsx:78]: what's the issue with [if not exists]?

**Review by joshuali925 — APPROVED** [review]

File: public/components/acceleration/visual_editors/query_builder.tsx (hunk 2 above)
Comment (ps48) [reply public/components/acceleration/visual_editors/query_builder.tsx:78]: This is not needed anymore by the backend.

File: public/components/acceleration/selectors/source_selector.tsx (hunk 1 above)
Comment (ps48) [reply public/components/acceleration/selectors/source_selector.tsx:81]: Not for this call, table name is loaded with this query

**Review by vamsi-amazon — APPROVED** [review]

**Pull request merged**